### PR TITLE
method POST building

### PR DESCRIPTION
**Pull request by PhonyFish**
created method post for new building and area

### CODE REVIEW

File: AC_Moderation_Backend/controllers/buildings_controllers.js [hunk 1]
```diff
@@ -4,14 +4,31 @@ const strings = require('../helpers/strings');
 async function postNewBuilding(req, res){
     new_building = new building();
     new_building.buildingLetter = req.body.buildingLetter;
-    new_building.subdepartments = req.body.subdepartments;
-    new_building.description = req.body.description;
+    //new_building.subdepartments = req.body.subdepartments;
```
Comment (Colima-97) [inline AC_Moderation_Backend/controllers/buildings_controllers.js]: Cut this line
```suggestion
```

File: AC_Moderation_Backend/controllers/buildings_controllers.js [hunk 2]
```diff
@@ -4,14 +4,31 @@ const strings = require('../helpers/strings');
...
 }
 
+async function postNewArea(req, res){
+    var letter = req.body.buildingLetter;
+    var cr = req.body.classroom;
+    var parent_exists = await building.exists({buildingLetter: letter});
+    if (parent_exists) {
+        var parent_building = building.findOne({buildingLetter: letter}, (err, done)=>{
+            if(err) return null
+            return done
```
Comment (Colima-97) [inline AC_Moderation_Backend/controllers/buildings_controllers.js]: So... We don't need to check anything, 'cause we already know that **parent_exists**.
I read in [StackOverflow](https://stackoverflow.com/questions/33383840/is-there-a-javascript-equivalent-for-the-python-pass-statement-that-does-nothing) that we can actually do nothing into an **if** statement.

```suggestion
            // Nothing to check here
```

Comment (PhonyFish) [reply AC_Moderation_Backend/controllers/buildings_controllers.js]: No, even if we don't use the if, the line of `return done` must be there. Or we could just delete the promise (is that what is called the (err, done)? ). But your suggestion isn't valid

Comment (Colima-97) [reply AC_Moderation_Backend/controllers/buildings_controllers.js]: I see...
Humm, then, we should check what **parent_building** is giving us. If it's **null** (Internet crashing or some stuff like that) the application will explode. Let's add an additional **if** statement to verify the content of **parent_building**.

Comment (PhonyFish) [reply AC_Moderation_Backend/controllers/buildings_controllers.js]: then we should let it stay like that because that's what it does

Comment (Colima-97) [reply AC_Moderation_Backend/controllers/buildings_controllers.js]: No, it's not. 
Whatever **parent_building** has, the code executes the line 24.

File: AC_Moderation_Backend/controllers/buildings_controllers.js [hunk 3]
```diff
@@ -4,14 +4,30 @@ const strings = require('../helpers/strings');
...
+        });
+        parent_building.areas.push({classroom: cr});
+        const saved = await parent_building.save();
+        if(saved) res.status(200).send({message: strings.Saved});
+        res.status(500).send({message: strings.err_500});
```
Comment (Colima-97) [inline AC_Moderation_Backend/controllers/buildings_controllers.js]:
```suggestion
        if(parent_building != null){
            parent_building.areas.push({classroom: cr});
            const saved = await parent_building.save();
            if(saved) res.status(200).send({message: strings.Saved});
            res.status(500).send({message: strings.err_500});
        }
```

**Pull request merged**